### DISCUSSION TITLE
petsc_options_iname and petsc_options_value

### DISCUSSION THREAD

**Discussion by wowodejiajia** [idaholab/moose#22704] [Q&A General] [answered]
Dear MOOSE experts,

I saw them in other programs
```
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'
```
and
```
  petsc_options_iname = '-pc_type -ksp_grmres_restart -sub_ksp_type
                         -sub_pc_type -pc_asm_overlap'
  petsc_options_value = 'asm      121                  preonly
                         lu          8'
```

I wrote a new input, but I don't know how to write them. Is there a detailed introduction to the above petsc_options_iname and petsc_options_value? 

Thank you.

**Comment by GiudGiud** [comment, marked as answer]
Hello

petsc_options_iname / value are options that get passed to petsc. They map to petsc options directly so the relevant documentation is the PETSc manual
https://petsc.org/release/docs/manual/

Guillaume

**Reply by wowodejiajia** [reply]
I'll see it. Thank you very much
